### ISSUE TITLE
feature request: setup compiler macros environment

### ISSUE THREAD

**Issue by privatenumber** [vuejs/eslint-plugin-vue#1662]
**Tell us about your environment**

- **ESLint version:** 7.32.0
- **eslint-plugin-vue version:** 7.19.1
- **Node version:** v14.18.1

**The problem you want to solve.**
It's [currently recommended to add macros to `globals`](https://github.com/vuejs/eslint-plugin-vue/blob/eaf6584/docs/user-guide/README.md#compiler-macros-such-as-defineprops-and-defineemits-are-warned-by-no-undef-rule) manually.

**Your take on the correct solution to problem.**

I noticed plugins can [export environments](https://eslint.org/docs/developer-guide/working-with-plugins#environments-in-plugins).

Wondering if eslint-plugin-vue can have a `setup-compiler-macros` environment that contains the 4 variables so we can do:

```js
{
	env: {
		'vue/setup-compiler-macros': true
	}
}
```

**Additional context**
